### PR TITLE
Refactor ZUIBadge

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR does some refactoring to the ZUIBadge component created by @KraftKatten 

## Changes
* Updates the styling of the text to match design 
* Adds option to show full or truncated number 
* Creates the `ActivityStatus` type and uses it in ZUIBadge and ZUIStatusChip

## Related issues
none

### CODE REVIEW

File: src/zui/ZUIBadge/index.tsx [hunk 1]
```diff
@@ -36,20 +44,30 @@ const useStyles = makeStyles<Theme, { status: ActivityStatus }>((theme) => ({
...
-  }
+  const showValue = number != undefined;
+  const style = showValue ? 'badge' : 'dot';
 
   return (
     <Box className={classes[style]}>
-      <Typography variant="labelSmMedium">{displayValue}</Typography>
+      <Typography className={classes.text}>
+        {showValue && !truncateLargeNumber && number.toString()}
+        {showValue && truncateLargeNumber && number > 99 && '99+'}
```
Comment (richardolsson) [inline src/zui/ZUIBadge/index.tsx:69]: This logic will render nothing if `truncateLargeNumber` is true but the number is less than 99.

**Review by richardolsson — CHANGES_REQUESTED** [review]
The code looks mostly good and the component seems to work as expected, except for a bug when the properties are configured just "right" (see below).

File: src/zui/ZUIBadge/index.tsx (hunk 1 above)
Comment (ziggabyte) [reply src/zui/ZUIBadge/index.tsx:69]: dammit i was staring at this thinking i had prevented just that. thank you for noticing!

**Review by richardolsson — COMMENTED** [review]
I forgot to approve before merging, but yeah, I approve! 😊

**Pull request merged**